quickwit/quickwit-search/src/root.rs

```diff
@@ -221,70 +250,97 @@ fn validate_request(
     Ok(())
 }
 
-/// Performs a distributed search.
-/// 1. Sends leaf request over gRPC to multiple leaf nodes.
-/// 2. Merges the search results.
-/// 3. Sends fetch docs requests to multiple leaf nodes.
-/// 4. Builds the response with docs and returns.
-#[instrument(skip(search_request, cluster_client, metastore))]
-pub async fn root_search(
+fn get_scroll_ttl_duration(search_request: &SearchRequest) -> crate::Result<Option<Duration>> {
+    let Some(scroll_ttl_secs) = search_request.scroll_ttl_secs else {
+        return Ok(None);
+    };
+    let scroll_ttl: Duration = Duration::from_secs(scroll_ttl_secs as u64);
+    if scroll_ttl > MAX_SCROLL_TTL {
+        return Err(SearchError::InvalidArgument(format!(
+            "Quickwit only supports scroll TTL period up to {} secs.",
+            MAX_SCROLL_TTL.as_secs()
+        )));
+    }
+    Ok(Some(scroll_ttl))
+}
+
+#[instrument(skip(search_request, cluster_client))]
+async fn search_partial_hits_round_with_scroll(
```

Comment:
```suggestion
async fn search_partial_hits_phase_with_scroll(
```

Reply:
It's common to hear about two-phase search or execution for search engines like ES or ours.

Reply:
ah yes!